internal/bminventory/inventory.go

```diff
@@ -6179,15 +6179,20 @@ func (b *bareMetalInventory) BindHostInternal(ctx context.Context, params instal
 		}
 	}
 
-	if !common.IsDay2Cluster(cluster) {
+	if !common.IsDay2Cluster(cluster) && host.FencingCredentials != "" {
 		fencingClustersSupported, fencingErr := common.BaseVersionGreaterOrEqual(common.MinimumVersionForTwoNodesWithFencing, cluster.OpenshiftVersion)
 		if fencingErr != nil {
 			return nil, common.NewApiError(http.StatusInternalServerError, fencingErr)
 		}
-		if host.FencingCredentials != "" && !fencingClustersSupported {
+		if !fencingClustersSupported {
 			err = errors.Errorf("Host %s has fencing credentials, it must be bound to a cluster with openshift version %s or newer", host.ID, common.MinimumVersionForTwoNodesWithFencing)
 			return nil, common.NewApiError(http.StatusBadRequest, err)
 		}
+		platform := getPlatformType(cluster.Platform)
```

Comment:
Can we add this validation to fencingClustersSupported and handle everything in one place? It's the same code as we have in updateHostFencing method